pkg/sshd/server.go

```diff
@@ -28,9 +30,65 @@ const (
 	// ServerConfig is the context key for ServerConfig object.
 	ServerConfig string = "ssh.ServerConfig"
 
-	multiplePush string = "Another git push is ongoing"
+	multiplePush       string = "Another git push is ongoing"
+	builderKeyLocation        = "/var/run/secrets/api/auth/builder-key"
 )
 
+// AuthKey authenticates based on a public key.
+func AuthKey(key ssh.PublicKey) (*ssh.Permissions, error) {
+	log.Info("Starting ssh authentication")
+	userInfo, err := controller.UserInfoFromKey(key)
+	if err != nil {
+		return nil, err
+	}
+
+	userInfo.Key = string(ssh.MarshalAuthorizedKey(key))
+	apps := strings.Join(userInfo.Apps, ", ")
+	log.Debug("Key accepted for user %s.", userInfo.Username)
+	perm := &ssh.Permissions{
+		Extensions: map[string]string{
+			"user":        userInfo.Username,
+			"fingerprint": userInfo.Fingerprint,
+			"apps":        apps,
+		},
+	}
+	return perm, nil
+}
+
+// Configure creates a new SSH configuration object.
+//
+// Config sets a PublicKeyCallback handler that forwards public key auth
+// requests to the route named "pubkeyAuth".
+//
+// This assumes certain details about our environment, like the location of the
+// host keys. It also provides only key-based authentication.
+// ConfigureServerSshConfig
+//
+// Returns:
+//  An *ssh.ServerConfig
+func Configure() (*ssh.ServerConfig, error) {
+	cfg := &ssh.ServerConfig{
+		PublicKeyCallback: func(m ssh.ConnMetadata, k ssh.PublicKey) (*ssh.Permissions, error) {
+			return AuthKey(k)
+		},
+	}
+	hostKeyTypes := []string{"rsa", "dsa", "ecdsa"}
+	pathTpl := "/var/run/secrets/deis/builder/ssh/ssh-host-%s-key"
+	for _, t := range hostKeyTypes {
+		path := fmt.Sprintf(pathTpl, t)
+
+		if key, err := ioutil.ReadFile(path); err == nil {
```

Comment:
or, if you want to skip host key parse errors, then `continue` in the second `if err != nil` block